### PR TITLE
fix(NODE-5000): make aws credential providers peer optional

### PR DESCRIPTION
**Pull request by nbbeeken**
#### What is changing?

aws credentials provider is now an optional peer dependency mirroring the driver (post v5) in order to remove the default installation of the optional package.

#### What is the motivation for this change?

AWS credential provider should only be installed when needed.

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat([NODE-1234](https://jira.mongodb.org/browse/NODE-1234))!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

**Comment by nbbeeken** [comment]
Driver patch: https://spruce.mongodb.com/version/63d15b7c3e8e860b62274c5c/tasks

File: bindings/node/etc/build-static.sh [hunk 1]
```diff
@@ -34,6 +33,4 @@ run_cmake --build . --target install --config RelWithDebInfo
 popd #./
 
 # build the `mongodb-client-encryption` addon
-# note the --unsafe-perm parameter to make the build work
```
Comment (baileympearson) [inline bindings/node/etc/build-static.sh:37]: Why did you remove `--unsafe-perm`?  Is it no longer relevant?

File: bindings/node/test/tools/utils.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+module.exports = Object.create(null);
```
Comment (baileympearson) [inline bindings/node/test/tools/utils.js:3]: We already have a file that builds a "requirements" object for tests to use (`requirements.helpers.js`).  Can we move this code there?

Also, following the pattern set in that file, it might be nice to simply export a boolean indicating whether or not the credential provider is installed (considering we'll be adding support for gcp / azure soon as well).

Maybe exporting something like this:
```javascript
{
  credentialProvidersInstalled: { 
    aws: <boolean>
  }
}
```

**Review by baileympearson — CHANGES_REQUESTED** [review]

File: bindings/node/etc/build-static.sh (hunk 1 above)
Comment (nbbeeken) [reply bindings/node/etc/build-static.sh:37]: Yep, the PR linked is closed, we also don't run as a root user in CI

**Review by baileympearson — APPROVED** [review]
